compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2129,10 +2129,10 @@ declare_lint! {
     /// [issue #51443]: https://github.com/rust-lang/rust/issues/51443
     /// [future-incompatible]: ../index.md#future-incompatible-lints
     pub WHERE_CLAUSES_OBJECT_SAFETY,
-    Warn,
+    Forbid,
```

Comment:
Most crater regressions are due to `deny(future_compat)` having bad interactions w `forbid`